source/nn/positional-encoding.lisp

```diff
@@ -1,6 +1,50 @@
+
+
 (in-package :caten/nn)
 
-;; from https://github.com/ml-explore/mlx/blob/main/python/mlx/nn/layers/positional_encoding.py
+; from https://github.com/ml-explore/mlx/blob/main/python/mlx/nn/layers/positional_encoding.py
 ;; RoPE
 ;; PositionalEncoding
 
+(defmodel (RoPE (dims &key (traditional NIL) (base 10000) (scale 1.0) (offset 1.0)))
+    ((dims dims)
+     (traditional traditional)
+     (base base)
+     (scale scale)
+     (offset offset)))
+
+(defmethod call ((op RoPE) &rest inputs)
+  (let* ((x (car inputs))
+         (shape (shape x))
+         (last-two (last shape 2))
+         (n (first last-two))     ; Sequence length
+         (d (second last-two))    ; Embedding size
+         (b (if (> (length shape) 2)
+                (reduce #'* (butlast shape 2))
+                1))               ; Batch size
+         (x (!reshape x (list b n d)))
+         (positions (!index-components (list n)))
+         (freqs (!exp (!div (!mul (!index-components (list (floor d 2)))
+                                  (!const x (- (log 10000))))
+                            (!const x d))))
+         (positions-reshaped (!reshape positions (list n 1)))    ; Shape: (n,1)
+         (freqs-reshaped (!reshape freqs (list 1 (floor d 2))))  ; Shape: (1, D/2)
+         (theta (!mul positions-reshaped freqs-reshaped))        ; Shape: (n, D/2)
+         (costheta (!cos theta))
+         (sintheta (!sin theta))
+         (x1 (!view x 't 't `(0 ,d 2)))  ; x[..., 0:d:2]
```

Comment:
t instead of 't. (the symbol t is preserved by cl so actually you do not have to quote it)